src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/legaladvisor/OrderCreator.java

```diff
@@ -220,8 +267,23 @@ private boolean hasRequestedForPermissionWithProvidedEvidence(CCDDirectionsQuest
             listItems.add(ImmutableMap.of(DYNAMIC_LIST_CODE, PilotCourtService.OTHER_COURT_ID,
                 DYNAMIC_LIST_LABEL, "Other Court"));
         }
+        Map<String, Object> hearingCourtListDefinition = new HashMap<>();
+        hearingCourtListDefinition.put(DYNAMIC_LIST_ITEMS, listItems);
+
+        if (StringUtils.isBlank(hearingCourtName)) {
+            return hearingCourtListDefinition;
+        }
+
+        Optional<Map<String, String>> selectedCourt =
+            listItems.stream().filter(s -> s.get(DYNAMIC_LIST_LABEL).equals(hearingCourtName)).findFirst();
+
+        if (selectedCourt.isPresent()) {
+            hearingCourtListDefinition.put(DYNAMIC_LIST_SELECTED_VALUE, selectedCourt.get());
+        } else if (pilot == Pilot.JDDO) {
+            hearingCourtListDefinition.put(DYNAMIC_LIST_SELECTED_VALUE, listItems.get(listItems.size() - 1));
```

Comment:
If we are after the Other Court, its better to put them as variable and return rather than returning size-1, Since if someone later adds the list, would not know about this implication.


ImmutableMap.of(DYNAMIC_LIST_CODE, PilotCourtService.OTHER_COURT_ID,
                DYNAMIC_LIST_LABEL, "Other Court"));	                DYNAMIC_LIST_LABEL, "Other Court")